add optional column indices to completecases

mkborregaard opened the pull request:
This is the most barebones possible implementation of the functionality discussed here: https://github.com/JuliaData/DataTables.jl/issues/38

mkborregaard committed:
add optional column indices to completecases

mkborregaard committed:
typo in doc string

nalimilan commented on src/abstractdatatable/abstractdatatable.jl:
````diff
@@ -437,12 +437,13 @@ end
 
 ```julia
-completecases(dt::AbstractDataTable)
+completecases(dt::AbstractDataTable[, indices])
````
`cols` rather.

mkborregaard replied:
done

nalimilan commented on src/abstractdatatable/abstractdatatable.jl:
```diff
@@ -468,6 +469,9 @@ function completecases(dt::AbstractDataTable)
 
+completecases(dt::AbstractDataTable, indices) = completecases(dt[indices])
+
+
```
A single line break is enough.

mkborregaard replied:
👍

nalimilan commented on src/abstractdatatable/abstractdatatable.jl:
```diff
@@ -437,12 +437,13 @@ end
 **Arguments**
 
 * `dt` : the AbstractDataTable
+* `indices` : which columns to check for `Null`s
```
Have a look at `nonunique` for an example of how it's usually documented, I think it's more detailed than that. Also need to say that all columns are used by default. For consistency wit that function, should also accept a single symbol or index.

mkborregaard replied:
Done. In fact I wrote the `cols` version of `nonunique` and the associated docstring, so I should have been in a position to get it right the first time around...

nalimilan reviewed (commented):
Thanks! It's always good to start with simple PRs rather than changing many things at the same time.

I think this change is uncontroversial, it makes `completecases` consistent with `nonunique`.

mkborregaard committed:
edits to completecases

Edited docstring to align with nonunique, and also implemented version when cols is not a Vector

nalimilan commented on src/abstractdatatable/abstractdatatable.jl:
```diff
@@ -438,11 +438,13 @@ Indexes of complete cases (rows without null values)
 **Arguments**
 
 * `dt` : the AbstractDataTable
+* `cols` : a column indicator (Symbol, Int, Vector{Symbol}, etc.) specifying the column(s) to check for Nulls
```
No uppercase for "null" AFAIK.

ararslan replied:
Should also apply code formatting to `Symbol`, `Int`, and `Vector{Symbol}`.

nalimilan replied:
That's actually an issue in the existing docstrings for similar functions. An additional commits to fix these would be welcome though.

nalimilan reviewed (commented):
Could you add tests for the new methods?

mkborregaard commented:
will do - should I update dropnull instead or in addition do you think?

nalimilan commented:
I would say in addition. There's also `dropnull!` for consistency.

nalimilan commented:
Bump.

mkborregaard commented:
I know! :-)
